test/secret/secret_test.go

```diff
@@ -0,0 +1,191 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package secret
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/splunk/splunk-operator/test/testenv"
+)
+
+var _ = Describe("secret test", func() {
+
+	var deployment *testenv.Deployment
+
+	BeforeEach(func() {
+		var err error
+		deployment, err = testenvInstance.NewDeployment(testenv.RandomDNSName(3))
+		Expect(err).To(Succeed(), "Unable to create deployment")
+	})
+
+	AfterEach(func() {
+		// When a test spec failed, skip the teardown so we can troubleshoot.
+		if CurrentGinkgoTestDescription().Failed {
+			testenvInstance.SkipTeardown = true
+		}
+		if deployment != nil {
+			deployment.Teardown()
+		}
+	})
+
+	Context("Standalone deployment (S1) with LM", func() {
+		It("secret: Secret update on a standalone instance", func() {
+
+			// Download License File
+			licenseFilePath, err := testenv.DownloadFromS3Bucket()
+			Expect(err).To(Succeed(), "Unable to download license file")
+
+			// Create License Config Map
+			testenvInstance.CreateLicenseConfigMap(licenseFilePath)
+
+			// Create standalone Deployment with License Master
+			standalone, err := deployment.DeployStandaloneWithLM(deployment.GetName())
+			Expect(err).To(Succeed(), "Unable to deploy standalone instance with LM")
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			// Modify Secret key with new value
+			secretKey := "hec_token"
+			modifiedKey := "6659B1AE-09B5-7101-530D-DA3FFD929501"
+			testenv.ModifySecretKey(deployment, testenvInstance.GetName(), secretKey, modifiedKey)
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			//Once system is up after update check each pod for secret key update
+			standaloneSecretName := fmt.Sprintf(testenv.SecretObjectStandalonePodName, deployment.GetName(), 2)
+			licenseMasterSecretName := fmt.Sprintf(testenv.SecretObjectLicenseMasterName, deployment.GetName(), 2)
+			monitoringConsoleSecretName := fmt.Sprintf(testenv.SecretObjectMonitoringConsoleName, testenvInstance.GetName(), 2)
+			verificationPods := []string{standaloneSecretName, licenseMasterSecretName, monitoringConsoleSecretName}
+
+			//Verify that each pod has the updated secret object
+			updated := testenv.VerifySecretObjectUpdate(deployment, testenvInstance, verificationPods, secretKey, modifiedKey)
+			Expect(updated).To(Equal(true), "All keys not updated")
```

Comment:
When does this print **"All keys not updated"**?

Reply:
When one of the keys is not updated on any of the objects

Reply:
This assertion should be moved into VerifySecretPodUpdate method